### ISSUE TITLE
Cannot initiate OTA-23 update on Volla Phone X due to missing /android/cache/recovery

### ISSUE THREAD

**Issue by Talkless** [ubports/ubuntu-touch#1999]
- Device: Volla Phone X 
- Channel: stable
- Build: OTA-22

### Steps to reproduce
Go to Settings -> Updates

### Expected behavior
Update button should work

### Actual behavior
"Try again" button is shown, with error message below:
> FileNotFoundError: [Errno 2] No such file or directory: '/android/cache/recovery'

### Logfiles and additional information
```
==> /home/phablet/.cache/upstart/application-legacy-ubuntu-system-settings-.log <==
void QSystemImage::downloadUpdate()
QQmlComponent: Component is not ready
qml: file:///usr/share/ubuntu/settings/system/qml-plugins/system-update/InstallationFailed.qml:-1 No such file or directory
qml: PopupUtils.open(): Failed to create the popup object.
```

```
$ ls -la /android/cache/
total 32
drwxr-xr-x.  3 root root  4096 May 28  2021 .
drwxr-xr-x. 23 root root  4096 Feb  9 16:52 ..
drwx------.  2 root root 16384 May 28  2021 lost+found
```

**Comment by koenvervloesem** [comment]
The same here. On https://t.me/utonvolla TheKit suggested to create the directory with `sudo mkdir /android/cache/recovery`. After a reboot the update button in Settings -> Updates seemed to work, as it started downloading OTA-23. But after clicking on the Install button, the device reboots into the recovery screen, giving me the choice Reboot, Factory reset and Advanced. When I choose Reboot, it just reboots into Ubuntu Touch, still OTA-22. I can again click on the update button, and the saga repeats itself.

**Comment by koenvervloesem** [comment]
In the mean time, Sveigder on the Telegram channel suggested this fix: https://github.com/ubports/ubports-installer/issues/2201#issuecomment-1034238633=

So after running `sudo mount -o bind /userdata/cache /android/cache` on the phone, I was able download and install OTA-23.

**Comment by Talkless** [comment]
Thanks @koenvervloesem , workaround worked!

Though after upgrade `/android/cache/recovery` is still missing, so the bug holds.

**Comment by OanaGS** [comment]
I wasn't able to perform any upgrade, I've got the error “android missing“ even in terminal...

**Comment by OanaGS** [comment]
I could finally upgrade to OTA-23, after 5 tries in a row  (reboots included). It's just Volla so bugy, or...?

**Comment by koenvervloesem** [comment]
> Though after upgrade `/android/cache/recovery` is still missing, so the bug holds.

After investigation, TheKit replied on Telegram:

> regarding OTA updates breakage on Volla X, the problem is having wrong version on mount-android.conf due to the bug with delta updates
> this should be the workaround

```shell
mount -o remount,rw /
wget https://gitlab.com/ubports/porting/community-ports/android9/volla-phone-x/volla-yggdrasilx/-/raw/halium-10.0/overlay/system/etc/init/mount-android.conf?inline=false -O /etc/init/mount-android.conf
reboot
```

**Comment by Talkless** [comment]
Looks like .conf is correct:
```
$ md5sum /etc/init/mount-android.conf 
41e239d619eac53c9fcb9929f9777f40  /etc/init/mount-android.conf
```
And now after reboot I do see `recovery`:
```
$ ls /android/cache/recovery/
last_locale
```

**Comment by Flohack74** [comment]
Please make sure you recreate the folder like this:
```
sudo mkdir /android/cache/recovery
sudo chmod 777 /android/cache/recovery
```
And then please reboot the device and search for updates once more.

**Comment by Silicomancer** [comment]
So what should an average user do? What is the official solution?

**Comment by Flohack74** [comment]
@Silicomancer this is the official solution, please try it out. We will add code in the next update that makes this impossible to happen again, but for now this manual step is needed so to make sure you can install an update again.

**Comment by Silicomancer** [comment]
> @Silicomancer this is the official solution, please try it out.

What exactly? This?

```
sudo mkdir /android/cache/recovery
sudo chmod 777 /android/cache/recovery
```

No `md5sum`? No `mount`?

**Comment by Flohack74** [comment]
@Silicomancer yes bascially the error is: its missing a directory. So, if you recreate this directory, all should be fine ;)

**Comment by Silicomancer** [comment]
> @Silicomancer yes bascially the error is: its missing a directory. So, if you recreate this directory, all should be fine ;)

Tried it and it doesn't work. Downloading now works. But after starting the installation the device restarts and then is stuck in the "UBports Recovery" screen.

**Comment by Flohack74** [comment]
@Silicomancer thats another problem then, and Volla support confirmed to me that there was a similar case. I am afraid you have to take the UBports installer now and reinstall latest version with it, try without checking bootstrap first. And never check wipe userdata, for obvious reasons ;)

**Comment by Silicomancer** [comment]
OMG, I hope this doesn't end with a brick. 
Is there a guide I need to follow or can I rely on the intuitive user guidance of the installer?

**Comment by Flohack74** [comment]
@Silicomancer its is indeed very hard to brick an Android device, as we do not reformat/repartition for flash bootloader code there is always a way out via fastboot :) The installer should have you covered.
